### PR TITLE
remove Operators module

### PR DESCRIPTION
**Pull request by JeffBezanson**
I have come to believe this module is at best useless. It's not a good practice to import a whole bunch of things at once and not be sure what you're getting, or what the code depends on. It's also not obvious what counts as an "operator"; at one time it may have been pretty clear but we have a lot more operators now, and have added some functions to the module that don't look like operators, so the classification is no longer useful.

Interestingly, while making this change I noticed that most of the Base modules that imported `Operators` had already switched to a different approach, either `import Base: +` or defining `Base.:(+)(...)`, such that the `importall Operators` line could just be deleted.

If this change is popular I can try to work out a deprecation mechanism.

### CODE REVIEW

**Comment by ararslan** [comment]
One possible deprecation strategy:

```julia
module Operators
    for op in [:!, :(!=), :(!==), :%, :&, :*, :+, :-, :/, ://, :<, :<:, :<<, :(<=),
               :<|, :(==), :(===), :>, :>:, :(>=), :>>, :>>>, :\, :^, :colon,
               :ctranspose, :getindex, :hcat, :hvcat, :setindex!, :transpose, :vcat,
               :xor, :|, :|>, :~, :×, :÷, :∈, :∉, :∋, :∌, :∘, :√, :∛, :∩, :∪, :≠, :≤,
               :≥, :⊆, :⊈, :⊊, :⊻, :⋅]
        @eval function $op(args...; kwargs...)
            depwarn("The Operators module is deprecated. Use Base.$op instead.", $op)
            Base.$op(args...; kwargs...)
        end
    end
end
```

The annoying part will be that the message will appear for every use of the imported operator...

**Comment by vtjnash** [comment]
+1. I've thought that the selection of names in this module was rather random anyhow.

**Comment by JeffBezanson** [comment]
Ok, should be ready to go.

**Review by ararslan — APPROVED** [review]

File: base/deprecated.jl [hunk 1]
```diff
@@ -1421,6 +1421,19 @@ function LibGit2.set_remote_url(path::AbstractString, url::AbstractString; remot
     LibGit2.set_remote_url(path, remote, url)
 end
 
+module Operators
+    for op in [:!, :(!=), :(!==), :%, :&, :*, :+, :-, :/, ://, :<, :<:, :<<, :(<=),
+               :<|, :(==), :(===), :>, :>:, :(>=), :>>, :>>>, :\, :^, :colon,
+               :ctranspose, :getindex, :hcat, :hvcat, :setindex!, :transpose, :vcat,
+               :xor, :|, :|>, :~, :×, :÷, :∈, :∉, :∋, :∌, :∘, :√, :∛, :∩, :∪, :≠, :≤,
+               :≥, :⊆, :⊈, :⊊, :⊻, :⋅]
+        if isdefined(Base, op)
```
Comment (tkelman) [inline base/deprecated.jl]: which aren't?

Comment (ararslan) [inline base/deprecated.jl]: `<|`

**Pull request merged**